src/operator/buffer.ts

```diff
@@ -23,6 +23,10 @@ export function buffer<T>(closingNotifier: Observable<any>): Observable<T[]> {
   return this.lift(new BufferOperator<T>(closingNotifier));
 }
 
+export interface BufferSignature<T> {
+  (closingNotifier: Observable<any>): Observable<T[]>;
```

Comment:
I think this is related to https://github.com/ReactiveX/RxJS/issues/1246. Probably we want to support return values other than observable - but I'd like to take care it as separate changes set include type update / test cases to ensure its behavior if necessary.